packages/blade/src/components/Box/BaseBox/getBaseBoxStyles.ts

```diff
@@ -1,102 +1,239 @@
 import type { CSSObject } from 'styled-components';
-import type { BaseBoxProps } from './types';
-import { getIn, makeBorderSize, makeSize, makeSpace } from '~utils';
+import type {
+  BaseBoxProps,
+  MakeValueResponsive,
+  SpacingValueType,
+  ArrayOfMaxLength4,
+} from './types';
+import type { Breakpoints } from '~tokens/global/breakpoints';
+import breakpoints from '~tokens/global/breakpoints';
+import { getMediaQuery } from '~src/utils/getMediaQuery';
+import { isReactNative, isEmpty, getIn, makeSpace, makeSize } from '~utils';
 import type { Theme } from '~components/BladeProvider';
 
-// allows 'auto' as spacing value
-const getSpacingValue = <SpacingValue extends string | number | undefined>({
-  value,
-  theme,
-}: {
-  value: SpacingValue;
-  theme: Theme;
-}): string | undefined => {
-  if (value === 'auto') return 'auto';
-  /**
-   * NOTE: don't allow numbers once we have the scale for sizing
-   * DO NOT PUBLISH THIS COMPONENT PUBLICLY WITH THIS PIECE OF CODE
-   */
-  if (typeof value === 'number') {
-    return makeSpace(value);
+const getResponsiveValue = <T extends string | number | string[]>(
+  value: MakeValueResponsive<T> | undefined,
+  size: keyof Breakpoints = 'base',
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): any | undefined => {
+  if (value === undefined || value === null) {
+    return undefined;
   }
-  return value ? makeSpace(getIn(theme, value)) : undefined;
+
+  if (typeof value === 'string' || typeof value === 'number' || Array.isArray(value)) {
+    if (size === 'base') {
+      return value;
+    }
+
+    // Plain values will already get added in base styles so we don't have to repeat them in media queries
+    return undefined;
+  }
+
+  if (isEmpty(value)) {
+    return undefined;
+  }
+
+  if (isReactNative()) {
+    // In React Native, we map the value `s` token on priority (since the breakpoint maps to mobiles in useBreakpoint hook).
+    // We further look into smaller sizes, then we check base size.
+    // Then we return the first non-undefined value in this priority
+    const priorityArray = [value.s, value.xs, value.base];
+    return priorityArray.find((val) => val !== undefined);
+  }
+
+  return value[size];
+};
+
+const getSpacingValue = (
```

Comment:
The reason I didn't turn it into object is because then calling the function gets too verbose.

```ts
getSpacingValue({ value: props.padding, theme: props.theme, size: screenSize }) // is a bit too verbose
getSpacingValue(props.padding, props.theme, screenSize) // this is less verbose
```

We're reusing this function a lot of times so better to keep things less verbose imo. I would've agreed with turning it into object if this was the function we were exposing to consumers. This is already used function several times so while adding new property, you would probably just refer to sequence from above line of code and use same.

Reply:
https://github.com/razorpay/blade/pull/1009#issuecomment-1447696941

Reply:
I think its still better making it an object otherwise it becomes way too prone to making mistakes